plugins/push/frontend/public/javascripts/countly.views.js

```diff
@@ -253,18 +256,234 @@
+        template: CV.T('/push/templates/push-notification-details-targeting-tab.html'),
+        data: function() {
+            return {
+                DAY_TO_MS_RATIO: 86400 * 1000
+            };
+        },
+        computed: {
+            pushNotification: function() {
+                return this.$store.state.countlyPushNotification.details.pushNotification;
+            }
+        },
+        methods: {
+            convertDaysInMsToDays: function(daysInMs) {
+                return daysInMs / this.DAY_TO_MS_RATIO;
+            }
+        }
+    });
+
+    var PushNotificationDetailsErrorsTab = countlyVue.views.create({
+        template: CV.T('/push/templates/push-notification-details-errors-tab.html'),
+        computed: {
+            errors: function() {
+                return this.$store.state.countlyPushNotification.details.pushNotification.errors;
+            },
+        },
+        methods: {
+            hasErrors: function() {
+                return this.$store.state.countlyPushNotification.details.pushNotification.failed > 0;
+            }
+        }
+    });
+
+    var MobileMessagePreview = countlyVue.views.create({
+        template: CV.T("/push/templates/mobile-message-preview.html"),
+        data: function() {
+            return {
+                selectedPlatform: this.getDefaultSelectedPlatform(),
+                PlatformEnum: countlyPushNotification.service.PlatformEnum,
+                appName: countlyGlobal.apps[countlyCommon.ACTIVE_APP_ID].name || CV.i18n('push-notification.mobile-preview-default-app-name')
+            };
+        },
+        props: {
+            platforms: {
+                type: Array,
+                default: []
+            },
+            title: {
+                type: String,
+                default: CV.i18n('push-notification.mobile-preview-default-title')
+            },
+            content: {
+                type: String,
+                default: CV.i18n('push-notification.mobile-preview-default-content'),
+            },
+            buttons: {
+                type: Array,
+                default: []
+            },
+            media: {
+                type: String,
+                default: null
+            }
+        },
+        watch: {
+            platforms: function() {
+                this.selectedPlatform = this.getDefaultSelectedPlatform();
+            }
+        },
+        methods: {
+            hasAndroidPlatform: function() {
+                return this.platforms.filter(function(platform) {
+                    return platform.value === countlyPushNotification.service.PlatformEnum.ANDROID;
+                }).length > 0;
+            },
+            hasIOSPlatform: function() {
+                return this.platforms.filter(function(platform) {
+                    return platform.value === countlyPushNotification.service.PlatformEnum.IOS;
+                }).length > 0;
+            },
```

Comment:
I'm not sure about the use case, but their name suggests that they would make a good fit for "computed" fields.

Reply:
actually their functionality as well. I will move them as computed properties.